src/engine/SCons/Spirent/__init__.py

```diff
@@ -4,11 +4,57 @@
 import json
 import os.path
 import time
+import subprocess
 
 import SCons.Action
 import SCons.Node
 import SCons.SConf
 
+def vulcan_builder(fs, options, graph):
+    # Get the vulcan command line options
+    vulcan_options = options.vulcan_opts
+    if not vulcan_options:
+        options_str = os.environ.get("SCONS_VULCAN_OPTS")
+        if not options_str:
+            print("scons: error, no options given to vulcan command")
+            return 1
+        else:
+            vulcan_options.append(options_str)
```

Comment:
This might also be simpler using `vulcan_options = options.vulcan_opts or os.environ.get(...)`. I'm not sure we need the error message here. And not sure about the `append` at line 22. Is `vulcan_options` going to be `None` before the `append`?

Reply:
I will remove the error message. vulcan.exe is throwing error when the command line options to it are given as string. I had an issue where the SCONS_VULCAN_OPTS set in the enviroment are not working but when given through --vulcan-opts it is working. The only difference is the earlier case it is string and later case it is list. 

vulcan_options will be empty list before append when --vulcan-opts is not given. So I am adding the SCONS_VULCAN_OPTS to it.